File: ios/Classes/Wrapper/Commands/InApp/InAppPauseCommand.swift
//
//  Created by Emarsys
//
import EmarsysSDK

public class InAppPauseCommand: EmarsysCommandProtocol {
    
    func execute(arguments: [String : Any]?, resultCallback: @escaping ResultCallback) {
        Emarsys.inApp.pause()
        resultCallback(["success": true])
    }
}

File: android/src/main/kotlin/com/emarsys/emarsys_sdk/command/inapp/InAppPauseCommand.kt
package com.emarsys.emarsys_sdk.command.inapp

import com.emarsys.Emarsys
import com.emarsys.emarsys_sdk.command.EmarsysCommand
import com.emarsys.emarsys_sdk.command.ResultCallback

class InAppPauseCommand: EmarsysCommand {
    override fun execute(parameters: Map<String, Any?>?, resultCallback: ResultCallback) {
        Emarsys.inApp.pause()
        resultCallback.invoke(null, null)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return true
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}